src/Host/Client/Impl/Host/RHost.BlobReadRequest.cs

```diff
@@ -0,0 +1,32 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.R.Host.Protocol;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.R.Host.Client {
+    partial class RHost {
+        private class BlobReadRequest : Request<byte[]> {
+            private BlobReadRequest(RHost host, Message message, CancellationToken cancellationToken)
+                : base(host, message, cancellationToken) {
+            }
+
+            public static Task<BlobReadRequest> ReadAllAsync(RHost host, ulong blobId, CancellationToken cancellationToken) {
+                return ReadAsync(host, blobId, 0, -1, cancellationToken);
```

Comment:
I prefer a strong indicator. Seeing a -1 in logs or while debugging makes the intent clear. Or a separate command for ReadAll.